src/NLog/StructuredEvents/TemplateEnumerator.cs

```diff
@@ -0,0 +1,381 @@
+                            case 4: return "4";
+                            case 5: return "5";
+                            case 6: return "6";
+                            case 7: return "7";
+                            case 8: return "8";
+                            case 9: return "9";
+                        }
+                        return parameterIndex.ToString(System.Globalization.CultureInfo.InvariantCulture);
+                    }
+                    else if (c == ' ')
+                    {
+                        SkipSpaces();
+                        c = Peek();
+                        if (c == '}' || c == ':' || c == ',')
+                            parameterIndex = parsed;
+                    }
+                }
+
+                _position = start;
+            }
+
+            return ReadUntil(HoleDelimiters);
+        }
+
+        /// <summary>
+        /// Parse format after hole name/index. Handle the escaped { and } in the format. Don't read the last }
+        /// </summary>
+        /// <returns></returns>
+        private string ParseFormat()
+        {
+
+            Skip(':');
+            string format = ReadUntil(TextDelimiters);
+            while (true)
+            {
+                var c = Read();
+
+                switch (c)
+                {
+                    case '}':
+                        {
+                            if (_position < _length && Peek() == '}')
+                            {
+                                //this is an escaped } and need to be added to the format.
+                                Skip('}');
+                                format += "}";
+                            }
+                            else
+                            {
+                                //done. unread the }
+                                _position--;
+                                //done
+                                return format;
+                            }
+                            break;
+                        }
+                    case '{':
+                        {
+                            //we need a second {, otherwise this format is wrong.
+                            var next = Peek();
+                            if (next == '{')
+                            {
+                                //this is an escaped } and need to be added to the format.
+                                Skip('{');
+                                format += "{";
+                            }
+                            else
+                            {
+                                throw new TemplateParserException($"Expected '{{' but found '{next}' instead in format.",
+                                    _position);
+                            }
+
+                            break;
+                        }
+                }
+
+                format += ReadUntil(TextDelimiters);
+            }
+        }
+
+        private int ParseAlignment()
```

Comment:
As we don't use this code for string.format, I think we could drop this. What do you think?

Reply:
No opinion. Just using your code